### ISSUE TITLE
Hooking into component lifecycle on the frontend

### ISSUE THREAD

**Issue by michaelpb** [adamghill/django-unicorn#251]
Hi,

I need to run some arbitrary JavaScript after every time a Unicorn component updates. This in order to do some recalculations due to potential reflows resulting from the DOM change of the Unicorn component. For my particular purpose, globally after ANY update on any component on the page, but per-component would also work (and I suspect there are situations where I would need one or the other).

I looked through the documentation, GitHub issues, and the source code, but to no avail. Can this be done?

-----

If there is not a way, I think the simplest solution would be to add some sort of "signal emitter" to the Unicorn object, or use the [JS event system](https://developer.mozilla.org/en-US/docs/Web/Events/Creating_and_triggering_events) on a Unicorn DOM element. Something like:

```javascript
// somewhere in my JS...
Unicorn.addEventListener('updated', () => console.log('got updated'));

// and then in the Unicorn code right after after morphdom is called:
Unicorn.dispatchEvent(new Event('updated'));
```

**Comment by adamghill** [comment]
This is a great idea! Do you think it makes sense to use a `CustomEvent` to include details with the component name? Then, you could just grab the event that makes sense at the time.

Interestingly, https://laravel-livewire.com/docs/2.x/lifecycle-hooks uses `Livewire.hook` for a similar purpose, but using the JS event system seems more straight-forward?

**Comment by adamghill** [comment]
@michaelpb https://github.com/adamghill/django-unicorn/pull/252 has an initial prototype, but I'm still working through naming and what the API should look like. Let me know if you have any feedback or thoughts.

**Comment by adamghill** [comment]
https://github.com/adamghill/django-unicorn/pull/252 is merged into master so this will be in 0.34.0. Closing.

**Comment by michaelpb** [comment]
Hey @adamghill -- pardon taking a second to get back to you on this (GitHub is too noisy sometimes), this looks exactly like what I was hoping for, thanks! I'll download it and give it a spin. Thanks again for this!!
